### PR TITLE
Fixed grammar in "Awaiting for signals" end note

### PR DESCRIPTION
**Pull request by Zshandi**
Added a few words to the note at the end of the "Awaiting for signals" section for proper grammar. Also changed "give" to "return" in the same note, so that it's more concise and consistent. This is also my first PR, so I hope I did it right!

### CODE REVIEW

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 1]
```diff
@@ -1771,8 +1771,8 @@ This also means that returning a signal from a function that isn't a coroutine w
         await get_signal()
         print("Button was pressed")
 
-.. note:: Unlike ``yield`` in previous Godot versions, you cannot obtain the function state object. This in spirit of
-          type-safety, because a function cannot say that returns an ``int`` but actually give a function state object
+.. note:: Unlike ``yield`` in previous Godot versions, you cannot obtain the function state object. This is in the spirit of
+          type-safety, because a function cannot say that it returns an ``int`` but actually return a function state object
```
Comment (Calinou) [inline tutorials/scripting/gdscript/gdscript_basics.rst:1775]:
```suggestion
.. note:: Unlike ``yield`` in previous Godot versions, you cannot obtain the function state object. This is done to ensure type safety.
A function cannot say that it returns an ``int`` while it actually returns a function state object
```

File: tutorials/scripting/gdscript/gdscript_basics.rst [hunk 2]
```diff
@@ -1771,8 +1771,8 @@ This also means that returning a signal from a function that isn't a coroutine w
         await get_signal()
         print("Button was pressed")
 
-.. note:: Unlike ``yield`` in previous Godot versions, you cannot obtain the function state object. This in spirit of
-          type-safety, because a function cannot say that returns an ``int`` but actually give a function state object
+.. note:: Unlike ``yield`` in previous Godot versions, you cannot obtain the function state object. This is done to ensure type safety.
+A function cannot say that it returns an ``int`` while it actually returns a function state object
```
Comment (Calinou) [inline tutorials/scripting/gdscript/gdscript_basics.rst:1775]:
```suggestion
.. note:: Unlike ``yield`` in previous Godot versions, you cannot obtain the function state object.
          This is done to ensure type safety.
          With this type safety in place, a function cannot say that it returns an ``int`` while it actually returns a function state object
```

**Review by Calinou — APPROVED** [review]
I made some changes to make the note more explicit (and more accessible to non-native English speakers), so I'll let someone else review before merging.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**